RIOT-OS/RIOT
# lwip: Bump version to v2.1.0

### Contribution description
As of about an hour ago [lwIP v2.1.0 is out](https://savannah.nongnu.org/forum/forum.php?forum_id=9248) and as we say in German "Ich hab da mal was vorbereitet…".

A reason why I want to have it in the upcoming release is because this has a chance of finally having working interoperability between lwIP's 6Lo and other 6Lo stacks.


### Testing procedure
Run `tests/lwip` on native and `at86rf2xx`-based nodes and try to exchange packets, run `tests/lwip_sock_*` with IPv4, IPv6, and both (set `LWIP_IPV4=1` or `LWIP_IPV6=1` respectively as environment variables) and see if they succeed.

### Issues/PRs references
Includes changes from https://github.com/RIOT-OS/RIOT/pull/8644 but is independent of that.
